File: app/src/main/java/com/mondiamedia/ahmedbadr/githubreopos/Application.java
package com.mondiamedia.ahmedbadr.githubreopos;

import com.mondiamedia.ahmedbadr.githubreopos.di.ApplicationComponent;
import com.mondiamedia.ahmedbadr.githubreopos.di.DaggerApplicationComponent;

import io.realm.Realm;
import io.realm.RealmConfiguration;

public class Application extends android.app.Application {

    public static final String NAME = "repos_db";
    public static final int VERSION = 0;
    public ApplicationComponent applicationComponent = DaggerApplicationComponent.create();

    @Override
    public void onCreate() {
        super.onCreate();
        initRealm();
    }

    private void initRealm() {
        Realm.init(this);
        RealmConfiguration config = new RealmConfiguration.Builder()
                .name(NAME)
                .schemaVersion(VERSION).deleteRealmIfMigrationNeeded()
                .build();

        Realm.setDefaultConfiguration(config);
    }
}

File: app/src/main/java/com/mondiamedia/ahmedbadr/githubreopos/Application.kt
package com.mondiamedia.ahmedbadr.githubreopos

import com.mondiamedia.ahmedbadr.githubreopos.di.ApplicationComponent
import com.mondiamedia.ahmedbadr.githubreopos.di.DaggerApplicationComponent

import io.realm.Realm
import io.realm.RealmConfiguration

class Application : android.app.Application() {
    var applicationComponent: ApplicationComponent = DaggerApplicationComponent.create()

    override fun onCreate() {
        super.onCreate()
        initRealm()
    }

    private fun initRealm() {
        Realm.init(this)
        val config = RealmConfiguration.Builder()
                .name(NAME)
                .schemaVersion(VERSION.toLong()).deleteRealmIfMigrationNeeded()
                .build()

        Realm.setDefaultConfiguration(config)
    }

    companion object {

        const val NAME = "repos_db"
        const val VERSION = 0
    }
}
